base/src/AffineTransform.cpp

```diff
@@ -0,0 +1,380 @@
+	NppiRect srcRect[4];
+	int srcPitch[4];
+	size_t srcNextPtrOffset[4];
+	NppiSize dstSize[4];
+	NppiRect dstRect[4];
+	int dstPitch[4];
+	size_t dstNextPtrOffset[4];
+
+	double shiftX;
+	double shiftY;
+	void *ctx;
+	NppStreamContext nppStreamCtx;
+};
+
+AffineTransform::AffineTransform(AffineTransformProps props) : Module(TRANSFORM, "AffineTransform", props)
+{
+	mDetail.reset(new Detail(props));
+}
+
+AffineTransform::~AffineTransform() {}
+
+bool AffineTransform::validateInputPins()
+{
+	if (getNumberOfInputPins() != 1)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateInputPins size is expected to be 1. Actual<" << getNumberOfInputPins() << ">";
+		return false;
+	}
+
+	framemetadata_sp metadata = getFirstInputMetadata();
+	FrameMetadata::FrameType frameType = metadata->getFrameType();
+	if (frameType != FrameMetadata::RAW_IMAGE && frameType != FrameMetadata::RAW_IMAGE_PLANAR)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateInputPins input frameType is expected to be RAW_IMAGE or RAW_IMAGE_PLANAR. Actual<" << frameType << ">";
+		return false;
+	}
+
+	FrameMetadata::MemType memType = metadata->getMemType();
+	if (memType != FrameMetadata::MemType::DMABUF)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateInputPins input memType is expected to be DMABUF. Actual<" << memType << ">";
+		return false;
+	}
+
+	return true;
+}
+
+bool AffineTransform::validateOutputPins()
+{
+	if (getNumberOfOutputPins() != 1)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateOutputPins size is expected to be 1. Actual<" << getNumberOfOutputPins() << ">";
+		return false;
+	}
+
+	framemetadata_sp metadata = getFirstOutputMetadata();
+	FrameMetadata::FrameType frameType = metadata->getFrameType();
+	if (frameType != FrameMetadata::RAW_IMAGE && frameType != FrameMetadata::RAW_IMAGE_PLANAR)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateOutputPins input frameType is expected to be RAW_IMAGE or RAW_IMAGE_PLANAR. Actual<" << frameType << ">";
+		return false;
+	}
+
+	FrameMetadata::MemType memType = metadata->getMemType();
+	if (memType != FrameMetadata::MemType::DMABUF)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateOutputPins input memType is expected to be DMABUF. Actual<" << memType << ">";
+		return false;
+	}
+
+	return true;
+}
+
+void AffineTransform::addInputPin(framemetadata_sp &metadata, string &pinId)
+{
+	Module::addInputPin(metadata, pinId);
+
+	mDetail->setMetadata(metadata);
+
+	mDetail->mOutputMetadata->copyHint(*metadata.get());
+	mDetail->mOutputPinId = addOutputPin(mDetail->mOutputMetadata);
+}
+
+bool AffineTransform::init()
+{
+	if (!Module::init())
+	{
+		return false;
+	}
+
+	return true;
+}
+
+bool AffineTransform::term()
+{
+	mDetail.reset();
+	return Module::term();
+}
+
+bool AffineTransform::process(frame_container &frames)
+{
+	auto frame = frames.cbegin()->second;
+	auto outFrame = makeFrame(mDetail->mFrameLength);
+	cudaFree(0);
+	cudaMemset(static_cast<DMAFDWrapper *>(outFrame->data())->getCudaPtr(), 0, outFrame->size());
```

Comment:
Change this to cudaMemsetAsync and pass the Cuda stream. Avoid using the default cuda stream. In Compute you are using stream so here also you can use the same